watertap/unit_models/nanofiltration_DSPMDE_0D.py

```diff
@@ -0,0 +1,1424 @@
+
+        ################################################################################################################
+        # Constraints
+        ################################################################################################################
+        if self.config.mass_transfer_coefficient == MassTransferCoefficient.spiral_wound:
+            # 0. Membrane area
+            @self.Constraint(doc="Membrane area")
+            def eq_area(b):
+                return b.area == b.length * b.width
+
+        # 1. Feed-solution/membrane equilibrium, DOF= Nj * 2 for inlet/outlet
+        @self.Constraint(self.flowsheet().config.time,
+                         io_list,
+                         phase_list,
+                         solute_set,
+                         doc="Interfacial partitioning at feed side of membrane")
+        def eq_interfacial_partitioning_feed(b, t, x, p, j):
+            return (b.pore_entrance[t, x].act_coeff_phase_comp[p, j] * b.pore_entrance[t, x].conc_mol_phase_comp[p, j]
+                    / (b.feed_side.properties_interface[t, x].act_coeff_phase_comp[p, j]
+                    * b.feed_side.properties_interface[t, x].conc_mol_phase_comp[p, j])
+                    ==
+                    b.partition_factor_steric_comp[t, j]
+                    * b.partition_factor_born_solvation_comp[t, j]
+                    * b.partition_factor_donnan_comp_feed[t, x, j]
+                    )
+
+        # 2. Permeate solution/membrane equilibrium, DOF= Nj * 2 for inlet/outlet
+        @self.Constraint(self.flowsheet().config.time,
+                         io_list,
+                         phase_list,
+                         solute_set,
+                         doc="Interfacial partitioning at permeate side of membrane")
+        def eq_interfacial_partitioning_permeate(b, t, x, p, j):
+            return (b.pore_exit[t, x].act_coeff_phase_comp[p, j] * b.pore_exit[t, x].conc_mol_phase_comp[p, j]
+                    / (b.permeate_side[t, x].act_coeff_phase_comp[p, j]
+                    * b.permeate_side[t, x].conc_mol_phase_comp[p, j])
+                    ==
+                    b.partition_factor_steric_comp[t, j]
+                    * b.partition_factor_born_solvation_comp[t, j]
+                    * b.partition_factor_donnan_comp_permeate[t, x, j])
+
+        # 3. Feed-solution/membrane electroneutrality, DOF=1 *2 for inlet/outlet: DOF= 2
+        @self.Constraint(self.flowsheet().config.time,
+                         io_list,
+                         phase_list,
+                         doc="Electroneutrality at feed-side membrane interface")
+        def eq_electroneutrality_interface(b, t, x, p):
+            return (sum(b.feed_side.properties_interface[t, x].conc_mol_phase_comp[p, j] *
+                    b.feed_side.properties_interface[t, x].charge_comp[j] for j in solute_set)
+                    == b.tol_electroneutrality)
```

Comment:
Since `tol_electroneutrality` is used in an equality constraint like this, I'm not sure "tolerance" is the right word. To me, this would be a proper tolerance if `tol_electroneutrality` were a `Var` with bounds `(-epsilon, epsilon)`. As it-is, it's more like an accumulated error term, and since it's used in a bunch of constraints, the accumulated error is probably different in all of them. Therefore I think having this as a parameter with a value *other than* 0 is a modeling error.

Another way to relax these constraints would be to apply a constraint scaling transform to the electroneutrality constraints, thereby relaxing them a bit (e.g., by scaling them by `1e-4`, for example).

Reply:
This is something I was planning to take out in the next PR.

Reply:
Maybe update the default value to be `0`?